app/main.py

```diff
@@ -1,6 +1,18 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> any:
```

Comment:
It is recommended to import `Any` from the `typing` module since it is not deprecated.